diff -u a/sic.c b/sic.c
--- a/sic.c	2013-05-01 11:58:37.610952607 +0200
+++ b/sic.c	2013-05-01 14:09:01.104435210 +0200
@@ -31,7 +31,7 @@
 	va_end(ap);
 	t = time(NULL);
 	strftime(timestr, sizeof timestr, "%D %R", localtime(&t));
-	fprintf(stdout, "%-12s: %s %s\n", channel, timestr, bufout);
+	fprintf(stdout, "\x1b[32m%-12s\x1b[0m: \x1b[33m%s\x1b[0m %s\n", channel, timestr, bufout);
 }
 
 static void
@@ -50,7 +50,7 @@
 		pout("", "No channel to send to");
 		return;
 	}
-	pout(channel, "<%s> %s", nick, msg);
+	pout(channel, "<\x1b[34m%s\x1b[0m> \x1b[31m%s\x1b[0m", nick, msg);
 	sout("PRIVMSG %s :%s", channel, msg);
 }
 
@@ -121,11 +121,11 @@
 	if(!strcmp("PONG", cmd))
 		return;
 	if(!strcmp("PRIVMSG", cmd))
-		pout(par, "<%s> %s", usr, txt);
+		pout(par, "<\x1b[34m%s\x1b[0m> %s", usr, txt);
 	else if(!strcmp("PING", cmd))
 		sout("PONG %s", txt);
 	else {
-		pout(usr, ">< %s (%s): %s", cmd, par, txt);
+		pout(usr, ">< \x1b[31m%s\x1b[0m (\x1b[32m%s\x1b[0m): %s", cmd, par, txt);
 		if(!strcmp("NICK", cmd) && !strcmp(usr, nick))
 			strlcpy(nick, txt, sizeof nick);
 	}
