diff -Naur atril-1.18.5.orig/shell/ev-window.c atril-1.18.5/shell/ev-window.c
--- atril-1.18.5.orig/shell/ev-window.c	2018-04-09 22:02:43.000000000 +1200
+++ atril-1.18.5/shell/ev-window.c	2018-09-01 20:56:33.798843740 +1200
@@ -1334,7 +1334,10 @@
 	if (!window->priv->metadata)
 		return;
 
-	if (ev_metadata_get_boolean (window->priv->metadata, "window_maximized", &maximized)) {
+	if (!(ev_metadata_get_boolean (window->priv->metadata, "window_maximized", &maximized))) {
+		gtk_window_maximize (GTK_WINDOW (window));
+		return;
+	} else {
 		if (maximized) {
 			gtk_window_maximize (GTK_WINDOW (window));
 			return;
