diff -Naur gthumb-3.2.8.orig/gthumb/gth-browser.c gthumb-3.2.8/gthumb/gth-browser.c
--- gthumb-3.2.8.orig/gthumb/gth-browser.c	2013-09-28 03:38:40.000000000 +1200
+++ gthumb-3.2.8/gthumb/gth-browser.c	2018-08-12 14:33:21.000000000 +1200
@@ -5408,11 +5408,15 @@
 		switch (event->keyval) {
 		case GDK_KEY_Page_Up:
 		case GDK_KEY_BackSpace:
+		case GDK_KEY_Left:
+		case GDK_KEY_Up:
 			gth_browser_show_prev_image (browser, FALSE, FALSE);
 			return TRUE;
 
 		case GDK_KEY_Page_Down:
 		case GDK_KEY_space:
+		case GDK_KEY_Right:
+		case GDK_KEY_Down:
 			gth_browser_show_next_image (browser, FALSE, FALSE);
 			return TRUE;
 
