diff -Naur gthumb-3.10.3.orig/gthumb/gth-application.c gthumb-3.10.3/gthumb/gth-application.c
--- gthumb-3.10.3.orig/gthumb/gth-application.c	2021-04-12 04:17:20.499223500 +1200
+++ gthumb-3.10.3/gthumb/gth-application.c	2021-08-26 22:19:23.691463998 +1200
@@ -146,7 +146,7 @@
 {
 	G_APPLICATION_CLASS (gth_application_parent_class)->startup (application);
 
-	g_object_set (gtk_settings_get_default (), "gtk-application-prefer-dark-theme", TRUE, NULL);
+	g_object_set (gtk_settings_get_default (), "gtk-application-prefer-dark-theme", FALSE, NULL);
 
 	gth_pref_initialize ();
 	migrate_data ();
