diff -Naur gthumb-3.2.8.orig/gthumb/gth-folder-tree.c gthumb-3.2.8/gthumb/gth-folder-tree.c
--- gthumb-3.2.8.orig/gthumb/gth-folder-tree.c	2013-05-01 04:56:12.000000000 +1200
+++ gthumb-3.2.8/gthumb/gth-folder-tree.c	2021-05-14 19:46:23.931335939 +1200
@@ -1075,13 +1075,12 @@
 
 static void
 _gth_folder_tree_add_empty_item (GthFolderTree *folder_tree,
-				 GtkTreeIter   *parent,
-				 gboolean       forced)
+				 GtkTreeIter   *parent)
 {
 	char        *sort_key;
 	GtkTreeIter  iter;
 
-	if (! forced && _gth_folder_tree_child_type_present (folder_tree, parent, ENTRY_TYPE_EMPTY))
+	if (_gth_folder_tree_child_type_present (folder_tree, parent, ENTRY_TYPE_EMPTY))
 		return;
 
 	sort_key = g_utf8_collate_key_for_filename (EMPTY_URI, -1);
@@ -1487,7 +1486,6 @@
 {
 	GtkTreeIter   parent_iter;
 	GtkTreeIter  *p_parent_iter;
-	gboolean      is_empty;
 	GHashTable   *file_hash;
 	GList        *scan;
 	GList        *old_files;
@@ -1507,8 +1505,8 @@
 	tree_model = GTK_TREE_MODEL (folder_tree->priv->tree_store);
 	gtk_tree_sortable_set_sort_column_id (GTK_TREE_SORTABLE (tree_model), GTK_TREE_SORTABLE_UNSORTED_SORT_COLUMN_ID, 0);
 
-	is_empty = TRUE;
-	_gth_folder_tree_add_empty_item (folder_tree, p_parent_iter, FALSE);
+	/* add the empty item first to not allow the folder to collapse. */
+	_gth_folder_tree_add_empty_item (folder_tree, p_parent_iter);
 
 	/* delete the children not present in the new file list, update the
 	 * already existing files */
@@ -1541,8 +1539,7 @@
 
 				if (g_hash_table_lookup (file_hash, file_data->file)) {
 					/* file_data is already present in the list, just update it */
-					if (_gth_folder_tree_set_file_data (folder_tree, &iter, file_data))
-						is_empty = FALSE;
+					_gth_folder_tree_set_file_data (folder_tree, &iter, file_data);
 					valid = gtk_tree_model_iter_next (tree_model, &iter);
 				}
 				else {
@@ -1571,10 +1568,8 @@
 	for (scan = files; scan; scan = scan->next) {
 		GthFileData *file_data = scan->data;
 
-		if (! g_hash_table_lookup (file_hash, file_data->file)) {
-			if (_gth_folder_tree_add_file (folder_tree, p_parent_iter, file_data))
-				is_empty = FALSE;
-		}
+		if (! g_hash_table_lookup (file_hash, file_data->file))
+			_gth_folder_tree_add_file (folder_tree, p_parent_iter, file_data);
 	}
 
 	_g_object_list_unref (old_files);
@@ -1582,8 +1577,7 @@
 
 	/**/
 
-	if (! is_empty)
-		_gth_folder_tree_remove_child_type (folder_tree, p_parent_iter, ENTRY_TYPE_EMPTY);
+	_gth_folder_tree_remove_child_type (folder_tree, p_parent_iter, ENTRY_TYPE_EMPTY);
 
 	if (p_parent_iter != NULL)
 		gtk_tree_store_set (folder_tree->priv->tree_store, p_parent_iter,
@@ -1671,7 +1665,6 @@
 {
 	GtkTreeIter  parent_iter;
 	GtkTreeIter *p_parent_iter;
-	gboolean     is_empty;
 	GList       *scan;
 
 	if (g_file_equal (parent, folder_tree->priv->root))
@@ -1681,19 +1674,14 @@
 	else
 		return;
 
-	is_empty = TRUE;
 	for (scan = files; scan; scan = scan->next) {
 		GthFileData *file_data = scan->data;
 
 		if (_gth_folder_tree_file_is_in_children (folder_tree, p_parent_iter, file_data->file))
 			continue;
-		if (_gth_folder_tree_add_file (folder_tree, p_parent_iter, file_data))
-			is_empty = FALSE;
+		_gth_folder_tree_add_file (folder_tree, p_parent_iter, file_data);
 	}
 
-	if (! is_empty)
-		_gth_folder_tree_remove_child_type (folder_tree, p_parent_iter, ENTRY_TYPE_EMPTY);
-
 	folder_tree->priv->recalc_entry_points = TRUE;
 }
 
@@ -1778,7 +1766,7 @@
 		return;
 
 	/* add the empty item first to not allow the folder to collapse. */
-	_gth_folder_tree_add_empty_item (folder_tree, p_parent_iter, TRUE);
+	_gth_folder_tree_add_empty_item (folder_tree, p_parent_iter);
 
 	for (scan = files; scan; scan = scan->next) {
 		GFile       *file = scan->data;
@@ -1788,8 +1776,7 @@
 			gtk_tree_store_remove (folder_tree->priv->tree_store, &iter);
 	}
 
-	if (gtk_tree_model_iter_n_children (GTK_TREE_MODEL (folder_tree->priv->tree_store), p_parent_iter) > 1)
-		_gth_folder_tree_remove_child_type (folder_tree, p_parent_iter, ENTRY_TYPE_EMPTY);
+	_gth_folder_tree_remove_child_type (folder_tree, p_parent_iter, ENTRY_TYPE_EMPTY);
 
 	folder_tree->priv->recalc_entry_points = TRUE;
 }
