diff -Naur gthumb-3.2.8.orig/gthumb/gth-file-list.c gthumb-3.2.8/gthumb/gth-file-list.c
--- gthumb-3.2.8.orig/gthumb/gth-file-list.c	2014-04-13 21:22:37.000000000 +1200
+++ gthumb-3.2.8/gthumb/gth-file-list.c	2021-05-07 16:41:03.374211727 +1200
@@ -1123,6 +1123,7 @@
 	gth_thumb_loader_set_max_file_size (file_list->priv->thumb_loader, g_settings_get_int (file_list->priv->settings, PREF_BROWSER_THUMBNAIL_LIMIT));
 	gth_file_selection_unselect_all (GTH_FILE_SELECTION (file_list->priv->view));
 
+	gth_file_view_set_vscroll (GTH_FILE_VIEW (file_list->priv->view), 0);	
 	gth_file_store_clear ((GthFileStore*) gth_file_view_get_model (GTH_FILE_VIEW (file_list->priv->view)));
 	g_hash_table_remove_all (file_list->priv->thumb_data);
 	gfl_add_files (file_list, files, -1);
